editor/Dockerfile

```diff
@@ -39,28 +41,30 @@ RUN echo $version > "$UNITY_PATH/version"
 #  Alias to unity-editor  #
 ###########################
 
-RUN echo '#!/bin/bash' > /usr/bin/unity-editor \
-  && chmod +x /usr/bin/unity-editor
+ADD editor/unity-editor /usr/bin/unity-editor
+RUN chmod 755 /usr/bin/unity-editor \
+  && mkdir /usr/bin/unity-editor.d
```

Comment:
Adding this external file and running different .sh files inside a folder is adding a lot of unnecessary complexity.

In my opinion the best way to solve the file overwrites would have been to simply move the last line of that file after the android clauses, and make sure that each addition is only added once.

I'm sorry about having to request changes again. It is however quite important that this Dockerfile stays very clean. As we publish thousands of images with it.

Reply:
But it also makes it useful for extending this script if someone needs to export few more variables in unity-editor script in another docker image which will be based on this editor.

Every /etc/profile does this with /etc/profile.d/*sh and many other scripts/configs use the same mechanism to combine various pieces together.

Reply:
This is out of scope for this change. 

There are much easier ways of adding variables.

When using Unity Builder, you can add it specifically for that workflow in its extended [Dockerfile](https://github.com/game-ci/unity-builder/tree/main/dist).

When not using Builder, it's still very easy to achieve by using

```dockerfile
FROM <this image:some-version-with-specific-compatibility-fixes-maintained-by-game-ci>

ENV my_var="my var"
```

Reply:
> I would argue that this is out of scope for this proposal.

Well, you wanted an elegant way in:

> What we need to make this PR merge-able is to find an elegant way account for multiple modules when creating that unity-editor alias script (and the .bashrc file).

and I believe these .d directories is quite well-established elegant way.

Being able to extend it from user images is just a small bonus (yes you can export variables with ENV, but you cannot do much more, like put something in  ANDROID_SDK_ROOT or use already exported variable (like ANDROID_SDK_ROOT, ANDROID_HOME, ANDROID_NDK_HOME) which will be available at runtime, but won't be in the extended Dockerfile.

If the main issue is the file being external to Dockerfile, I can create it with here-doc inside Dockerfile, it's just a bit more error prone with Dockerfile syntax getting in a way of simple here-doc.

Reply:
I don't necessarily think it's a bad way of doing things, but it does increase complexity.

In my opinion my proposed solution of "simply moving the last line of the unity-editor alias file after the android clauses, and make sure that each addition is only added once" is much simpler and more appropriate for the scope of this proposal, which is about supporting multiple modules - not finding the best way to make images extensible.

That said, lets get some additional opinions in here and decide based on what they say.

cc: @davidmfinol @frostebite @GabLeRoux

Reply:
I tend to agree with @webbertakken and his explanation here: https://github.com/game-ci/docker/pull/116#discussion_r641826241

> /etc/profile does this with /etc/profile.d/*sh

Indeed, but as part of the docker image here, I'm not sure I would have had the reflex to look in that folder.

Using a here-doc would be a bit more verbose indeed, but if the goal is to allow user to extend the image and if _moving the last line of that file after the android clauses_ solves that, I would go with @webbertakken's suggestion.